### DISCUSSION TITLE
How to extract port locations from one design and impliment on another?

### DISCUSSION THREAD

**Discussion by Garpurinn** [ansys/pyaedt#3481] [Q&A]
Hi,

I am having difficulties finding a good way to realize the following script: 
1) Extract locations of ports on an older design (for example from an EDB export of an older SIwave project) and store it in a list.
2) In a new layout design (with same pad locations) create the same ports using the information from the list. 

I have managed to make use of the following to find the ports on the older design:
`# Find all the ports in the project `
`PortsFound = edb.siwave.excitations`

This gives me a dictionary, were the "keys" are the names of the ports. But in the "values" fields I get some data that I am not sure how I can use to recreate the port location - like this:
`<pyaedt.edb_core.edb_data.sources.ExcitationPorts object at 0x000002A97FA0A800>`

Can anybody help me connect the dots from here towards duplicating the port onto a new design? Or is `edb.siwave.excitations` maybe not the correct start for acquiring port locations? 

Kind regards,
Garpurinn

**Comment by Samuelopez-ansys** [comment]
@svandenb-dev

**Comment by svandenb-dev** [comment]
Hi @Garpurinn,
I have added a new property when you create ports on padstacks. 
edb = Edb(edbpath=r"C:\Temp\edb_ports\test.aedb")
terms = list(edb.terminals.values())
terms[0].position
This is not available with PointTerminal or Pingroup.
